### ISSUE TITLE
Validation with the Data Annotation Validators - ASP.NET MVC

### ISSUE THREAD

**Issue by vijayantkatyal** [Knockout-Contrib/Knockout-Validation#220]
e.g I have this model

public class Product
{
        public int Id { get; set; }

```
    [Required]
    [StringLength(10)]
    public string Name { get; set; }

    [Required]
    public string Description { get; set; }

    [DisplayName("Price")]
    [Required]
    [RegularExpression(@"^\$?\d+(\.(\d{2}))?$")]
    public decimal UnitPrice { get; set; }
```

}

how to map the validation attributes to knockout viewmodel

solution can be support of jquery unobtrusive validation

<input class="text-box single-line" 
    data-val="true" data-val-number="The field Qty Available must be a number." 
    data-val-range="The field Qty Available must be between 0 and 120." 
    data-val-range-max="120" data-val-range-min="0" 
    data-val-required="The Qty Available field is required." 
    id="QtyOnHand" name="QtyOnHand" type="text" value="12" />

I can generate about html markup manually or using html helpers , the point is i only have to write validation on back-end model.

in other words validation rules on html markup instead on java-script code.

or is there any other approach doing it correctly ?
